content/client-lib-development-guide/features.textile

```diff
@@ -459,7 +459,7 @@ h3(#realtime-channel). Channel
 ** @(RTL3b)@ If the connection state enters the @CLOSED@ state, then an @ATTACHING@ or @ATTACHED@ channel state will transition to @DETACHED@
 ** @(RTL3c)@ If the connection state enters the @SUSPENDED@ state, then an @ATTACHING@ or @ATTACHED@ channel state will transition to @SUSPENDED@
 ** @(RTL3d)@ If the connection state enters the @CONNECTED@ state, then a @SUSPENDED@ channel will initiate an attach operation and transition to @ATTACHING@. If the attach operation times out, the channel should return to the @SUSPENDED@ state (see "RTL4f":#RTL4f).
-* @(RTL11)@ If a channel enters the @DETACHED@, @SUSPENDED@ or @FAILED@ state, then all messages that are still queued for send on that channel should be deleted from the queue triggering a failure for the publish or presence methods invoked for those messages
+* @(RTL11)@ If a channel enters the @SUSPENDED@ or @FAILED@ state, then all messages that are still queued for send on that channel should be deleted from the queue triggering a failure for the publish or presence methods invoked for those messages
```

Comment:
So: this spec item is now only useful for PresenceMessages, as Messages are no-longer per-channel queued. (Don't think we need it for connection-queued messages -- a channel isn't going enter the `suspended` or `failed` state when the connection isn't `connected` except as a result of the connection entering the `suspended` or `failed` states, at which point connection-queued messages will be failed anyway). But for PresenceMessages this change is wrong, they should be failed when the channel is detached.

How about:
`* @(RTL11)@ If a channel enters the @DETACHED@, @SUSPENDED@ or @FAILED@ state, then all presence actions that are still queued for send on that channel per "RTP16b":#RTP16b should be deleted from the queue, and any callback passed to the corresponding presence method invocation should be called with an @ErrorInfo@ indicating the failure`

Reply:
> So: this spec item is now only useful for PresenceMessages, as Messages are no-longer per-channel queued. 

Yes, I suppose that is correct.  

Odd that we now have two mechanisms for delivering messages on channels, but it's necessary.

@paddybyers you happy with @SimonWoolf's suggested change?

Reply:
> @paddybyers you happy with @SimonWoolf's suggested change?

lgtm